Test branch

Opkolopukos opened the pull request:
Gradle build

Opkolopukos committed:
Created method that merges two arrays and writes result in a new txt File.

Opkolopukos committed:
Changed method arguments to n-Lists.

Opkolopukos committed:
test:
Changed method arguments to listOfLists.
Added mergeUtils class that includes listOfLists merging method.
Separated Sorter and MergeUtils classes.

Opkolopukos committed:
- Created inputFiles class which processes input files and opens reading streams.
- Changed project format to Gradle (from Maven).

Opkolopukos committed:
- Created inputFiles class which processes input files and opens reading streams.
- Changed project format to Gradle (from Maven).

Opkolopukos committed:
Merge remote-tracking branch 'origin/test' into test

Opkolopukos committed:
- Added ascending merge sort for files with Strings.
test: - Tried to implement generics.

pvlnsk commented on MergeSort.iml:
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
лишнее

pvlnsk commented on gradlew:
```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env sh
```
можешь инфраструктурные файлы залить прямиком мастер, тогда их не будет в дифе

Opkolopukos committed:
Депрессивный комит. Не получается компаратор прикрутить, эти узлы должны быть комперебл.

Не получается читать из каждого файла с итерированием первые строки и их добавлять в узлы - сравнивая, записывая нужное значение в файл и сохраняя остальные.

Opkolopukos committed:
Депрессивный комит. Не получается компаратор прикрутить, эти узлы должны быть комперебл.

Не получается читать из каждого файла с итерированием первые строки и их добавлять в узлы - сравнивая, записывая нужное значение в файл и сохраняя остальные.

pvlnsk reviewed (commented):
Забыл отправить комментарии раньше

pvlnsk commented:
Ты можешь считать из файла строку и сложить её в поле класса? Сделай это

Opkolopukos committed:
Добавил простой парсер командной строки.
Добавил обратную сортировку.

Сортировка работает при запуске с параметрами командной строки.
Проблемы указаны в fixme в мейне.

Opkolopukos committed:
Добавил класс для отдельного инпут файла, позволяющий читать и сохранять текушую строку из файла.

Opkolopukos commented:
> Ты можешь считать из файла строку и сложить её в поле класса? Сделай это

Сделано. Отдельный класс читающий и запоминающий строки.

Opkolopukos committed:
Класс ПсевдоМёрж с псевдокодом

Commit:
fasdfd

Opkolopukos committed:
Рабочий вариант сортировки с парсингом аргументов из командной строки.

Opkolopukos committed:
Merge remote-tracking branch 'origin/test' into test

# Conflicts:
#	src/main/java/merging/MergeUtils.java
#	src/main/java/merging/PseudoMerge.java

Opkolopukos committed:
Добавил проверку на совпадение порядка сортировки в параментрах и порядка сортировки входных файлов.
+ Проверка на пробел в строке.

Opkolopukos committed:
Добавил частичную сортировку входных файлов. Если елемент не соотстветствует логике заданного компаратора - ищется подходящий элемент до конца файла.
Добавил уведомление в консоль на случай невозможности парсинга входных файлов по заданному параметру (пример. инпут строк + параметр сортировки чисел)
Добавил проверку для парсера командной строки на максимально допустимое количество параметров + изменения уведомлений в случае некорректных параметров.

Opkolopukos committed:
Update README.md

Opkolopukos committed:
Gradle built, final version
All I/O streams put into resources.
Moved merge logic to client from CMDParser class.

Opkolopukos committed:
Merge remote-tracking branch 'origin/test' into test

# Conflicts:
#	README.md

Opkolopukos committed:
Merge branch 'master' into test
